crypto/ec/ec_ameth.c

```diff
@@ -568,7 +597,9 @@ const EVP_PKEY_ASN1_METHOD eckey_asn1_meth = {
 
     0, 0, 0,
 
-    ec_pkey_check
+    ec_pkey_check,
+    ec_pkey_public_check,
+    ec_pkey_param_check
```

Comment:
Please implement this for the other algos that have an `EVP_PKEY_ASN1_METHOD` as well...

Reply:
Okay. Let me figure out... seems this needs some efforts...

Reply:
The following algorithms have `EVP_PKEY_ASN1_METHOD`: MAC(cmac, hmac, poly1305, siphash), DH, DSA, ECC, RSA. Since these `check` methods are used for public-key algorithms, so the MAC algos should be ignored.

Reply:
@levitte Right?

Reply:
Yes.  Note also that DH and DSA don't even have `pkey_check` defined, so I'll be satisfied with a completed RSA implementation.

Reply:
> so I'll be satisfied with a completed RSA implementation.

Errr, I don't quite get this. RSA also doesn't have `public_check` and `param_check` methods due to characters of the algorithm.

Reply:
But I am going to implement DH and DSA's checking methods

Reply:
Re RSA, you lost me completely...  how can RSA support `public_check` and `param_check` when you haven't touched `crypto/rsa/rsa_ameth.c`?????

Reply:
What?

Since RSA has no way to support `public_check` and `param_check`, so just leave the two fields of `EVP_PKEY_ASN1_METHOD` as `0` in `crypto/rsa/rsa_ameth.c` is fine enough. That file should not be touched in this pull request anyway...